File: 088_merge_sorted_array/Solution.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int idx = m-- + n-- - 1;
        while (m >= 0 && n >= 0) {
            nums1[idx--] = nums1[m] > nums2[n] ? nums1[m--] : nums2[n--];
        }

        while (n >= 0) nums1[idx--] = nums2[n--];
    }

    public static void main(String[] args) {
        int[] nums1 = {1, 2, 3, 0, 0, 0};
        int[] nums2 = {2, 5, 6};
        int m = 3, n = 3;

        new Solution().merge(nums1, m, nums2, n);
        for (int num : nums1) {
            System.out.print(num + " ");
        }
    }
}

File: 088_merge_sorted_array/Solution.py
class Solution:
    def merge(self, nums1: list, m: int, nums2: list, n: int) -> None:
        idx = m + n - 1
        m, n = m - 1, n - 1
        while m >= 0 and n >= 0:
            if nums1[m] > nums2[n]:
                nums1[idx] = nums1[m]
                idx, m = idx - 1, m - 1
            else:
                nums1[idx] = nums2[n]
                idx, n = idx - 1, n - 1

        while n >= 0:
            nums1[idx] = nums2[n]
            idx, n = idx - 1, n - 1


if __name__ == '__main__':
    nums1 = [1, 2, 3, 0, 0, 0]
    nums2 = [2, 5, 6]
    m, n = 3, 3
    Solution().merge(nums1, m, nums2, n)
    print(nums1)

